pandas/_typing.py

```diff
@@ -222,6 +229,7 @@ def __reversed__(self) -> Iterator[_T_co]:
 
 # to maintain type information across generic functions and parametrization
 T = TypeVar("T")
+P = ParamSpec("P")
```

Comment:
probably needs to move inside the TYPE_CHECKIG block as it is None at runtime (see failing tests).

Reply:
Yeah I'm still playing with how to structure this. If we move this fully inside `TYPE_CHECKING`, then we'll also have to import it in other files only under `TYPE_CHECKING` to prevent import errors, so going to think a little in how to structure this so it's not as complicated to import in other files, if possible.

Reply:
You could define `P = None` at runtime

Reply:
[Went with a version of this](https://github.com/pandas-dev/pandas/pull/56760/commits/6155d05772c89ac47984a02e68812912fca23a37#diff-80af4c7d6ece8cfeec6401a5f5babe27f19a7fc7476ad9148dca02f58a07a8dcR103-R112)!